Client/Ecosia/UI/MyImpact/EcosiaHome.swift

```diff
@@ -323,7 +323,9 @@ final class EcosiaHome: UICollectionViewController, UICollectionViewDelegateFlow
 
         switch section {
         case .impact, .legacyImpact:
-            return CGSize(width: view.bounds.width - 2 * margin, height: 226)
+            let dynamicTypeResize = max(UIFont.preferredFont(forTextStyle: .body).pointSize - 17, 0) * 7
```

Comment:
oh.. quite an equation here 🤓 I'm wondering how that fixes it as the the return value of this method is only an estimation and the real size is determined via self-sizing 🤔 Though it's a tricky cell with nested StackViews, so I assume this solution her is easier than tinkering around in StackView margins 👍

Reply:
If I'm not mistaken the StackView is self sizing, but the cell itself is not. So the problem was that the StackView was overflowing the cell (and it looked closed to the separator than what it really was). I don't think there is something as dynamic cells for a CollectionView.

The other solution I have seen used is to determine the size it will require even before displaying the CollectionView itself, but that implies a whole different process, although we can explore it if we see this doesn't fix the problem.

Reply:
@vauxhall Yes there are self-sizing collectionview cells and we make us of it. Check out https://developer.apple.com/documentation/uikit/uicollectionviewflowlayout/1779556-automaticsize

It's not a really intuitive API and has some drawbacks and caveats. E.g. to make it work we need to set the the width of the cells manually via a constraint and then they self-size (theoretically ;) )